crates/host/src/wasmbus/mod.rs

```diff
@@ -978,14 +985,79 @@ impl Host {
             ctl_nats_url = config.ctl_nats_url.as_str(),
             "connecting to NATS control server"
         );
-        let nats = async_nats::connect(config.ctl_nats_url.as_str())
-            .await
-            .context("failed to connect to NATS control server")?;
+        let ctl_nats = match (config.ctl_jwt.as_ref(), config.ctl_seed.as_ref()) {
```

Comment:
Since this logic is duplicated for ctl, rpc, and prov_rpc, can we extract to a function?